sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs

```diff
@@ -18,42 +15,28 @@ namespace Azure.Data.Tables
     /// </summary>
     public class TableTransactionalBatch
     {
-        private readonly TableRestClient _tableOperations;
-        private readonly TableRestClient _batchOperations;
-        private readonly string _table;
-        private readonly ClientDiagnostics _diagnostics;
-        private MultipartContent _changeset;
-        private readonly OdataMetadataFormat _format;
-        private readonly ResponseFormat _returnNoContent = ResponseFormat.ReturnNoContent;
+        private readonly SemaphoreSlim _batchLock = new(1, 1);
+        internal MultipartContent _changeset;
         internal MultipartContent _batch;
-        internal Guid _batchGuid;
-        internal Guid _changesetGuid;
-        internal ConcurrentDictionary<string, (HttpMessage Message, RequestType RequestType)> _requestLookup = new ConcurrentDictionary<string, (HttpMessage Message, RequestType RequestType)>();
-        internal ConcurrentQueue<(ITableEntity Entity, HttpMessage HttpMessage)> _requestMessages = new ConcurrentQueue<(ITableEntity Entity, HttpMessage HttpMessage)>();
-        private List<(ITableEntity Entity, HttpMessage HttpMessage)> _submittedMessageList;
-        private bool _submitted;
+        private Guid _batchGuid;
+        private Guid _changesetGuid;
+        internal readonly Dictionary<string, BatchItem> _requestLookup = new();
+        internal ConcurrentQueue<string> _requestMessages = new();
+        internal List<BatchItem> _submittedMessageList;
+        internal bool _submitted;
```

Comment:
I think this may benefit from being `volatile` given its usage in the client when sending.